packages/aws-cdk/lib/api/bootstrap-environment.ts

```diff
@@ -7,18 +7,44 @@ import { SDK } from './util/sdk';
 export const BUCKET_NAME_OUTPUT = 'BucketName';
 export const BUCKET_DOMAIN_NAME_OUTPUT = 'BucketDomainName';
 
-export async function bootstrapEnvironment(environment: Environment, aws: SDK, toolkitStackName: string, roleArn: string | undefined, toolkitBucketName: string | undefined): Promise<DeployStackResult> {
+/**
+ * Possible customizations for the CDK Bootstrap stack.
+ */
+export interface BootstrapEnvironmentProps {
+  /**
+   * The name to be given to the CDK Bootstrap bucket.
+   *
+   * @default a name is generated by CloudFormation.
+   */
+  readonly bucketName?: string;
+
+  /**
+   * The ID of an existing KMS key to be used for encrypting items in the bucket.
+   *
+   * @default the default KMS key for S3 will be used.
+   */
+  readonly kmsKeyId?: string;
+}
+
+export async function bootstrapEnvironment(environment: Environment, aws: SDK, toolkitStackName: string, roleArn: string | undefined, props: BootstrapEnvironmentProps = {}): Promise<DeployStackResult> {
   const synthesizedStack: SynthesizedStack = {
     environment,
     metadata: {},
     template: {
       Description: "The CDK Toolkit Stack. It was created by `cdk bootstrap` and manages resources necessary for managing your Cloud Applications with AWS CDK.",
       Resources: {
         StagingBucket: {
-          Type: "AWS::S3::Bucket",
+          Type: 'AWS::S3::Bucket',
           Properties: {
-            AccessControl: "Private",
-            BucketEncryption: { ServerSideEncryptionConfiguration: [{ ServerSideEncryptionByDefault: { SSEAlgorithm: "aws:kms" } }] }
+            AccessControl: 'Private',
```

Comment:
@hoegertn there is no condition here for when no `kmsKeyId` is provided.

Reply:
right, you'd want to default that to a KMS managed server-side encryption